src/Analyzers/CSharp/Analyzers/UseImplicitObjectCreation/CSharpUseImplicitObjectCreationDiagnosticAnalyzer.cs

```diff
@@ -93,6 +94,15 @@ private void AnalyzeSyntax(SyntaxNodeAnalysisContext context)
                     _ => null,
                 };
             }
+            else if (objectCreation.Parent.IsKind(SyntaxKind.ArrayInitializerExpression, SyntaxKind.CollectionInitializerExpression))
+            {
+                typeNode = objectCreation.Parent.Parent switch
+                {
+                    ArrayCreationExpressionSyntax arrayCreation => arrayCreation.Type.ElementType,
+                    ObjectCreationExpressionSyntax { Type: QualifiedNameSyntax { Right: GenericNameSyntax { TypeArgumentList: TypeArgumentListSyntax argumentList } } } => argumentList.Arguments.Count == 1 ? argumentList.Arguments[0] : null,
+                    _ => null,
+                };
+            }
```

Comment:
break these into two else-if cases.  i don't see the benefit of checking for two kinds, but then having two separate switch cases for each of those.

Reply:
I thought they're conceptually very similar syntax so I did them in one block because of that. But noted, will do.